tidy3d/components/simulation.py

```diff
@@ -436,6 +437,21 @@ def _warn_monitor_simulation_frequency_range(cls, val, values):
                 )
         return val
 
+    @pydantic.validator("monitors", always=True)
+    def n2f_monitors_with_symmetry(cls, val, values):
+        """Error if a server-side near-to-far monitor is used with symmetry"""
+        monitors = val
+        symmetry = values.get("symmetry")
+        if not any(symmetry):
+            return val
+        for monitor in monitors:
+            if isinstance(monitor, AbstractNear2FarMonitor):
+                raise SetupError(
+                    "Server-side near-to-far calculation is not available with symmetry. "
+                    "Use the local near-to-far available through the ``RadiationVectors`` class."
+                )
+        return val
```

Comment:
This is technically only a problem if the n2f monitor is intersecting any of the symmetry planes. There's a related validator for mode objects (there, it's also allowed for the object to be centered on the symmetry center). https://github.com/flexcompute/tidy3d/blob/da74c98eeefec83919e7172098ecab801f4598b8/tidy3d/components/validators.py#L67-L95

Reply:
Even if the monitor isn't crossing a symmetry plane, isn't there still some ambiguity in the setup? I.e., one might assume that there is a "mirror" monitor in the symmetry region, and the total far field is the result of surface currents on both monitors?

Reply:
The approach I have been taking is that the simulation setup is agnostic of the defined symmetries. The symmetries only work to tell the solver how to speed things up, but apart from that your sources and monitors work exactly the same way whether you defined symmetries or not. So in your example, your single monitor will return the same result with or without a defined symmetry, and if you wanted to sum things up, you just put two monitors symmetrically (as you would if you hadn't explicitly defined symmetries), and we handle everything on the backend. This is already handled like this for FieldMonitor and FluxMonitor-s, and ModeMonitor's are required to be either in the main symmetry quadrant or sitting on the symmetry plane (more because of integrating them with the solver than anything else). But you can e.g. place a FieldMonitor at x = -1 for a symmetry at x = 0 and it should still work even though the main quadrant (where the fields are actually stored) is at x >=0. That said, I think the check for server side far field monitors for now should be that they are entirely in the main quadrant, before we figure out how to handle symmetries better for those.